common.py

```diff
@@ -400,54 +399,77 @@ def prepare_filelist(self, names, path, goto, indent):
         items += files
         return items
 
-    def is_hidden(self, filename, path, goto=''):
+    def is_hidden(self, item, path, goto='', get_is_dir=False, dirs_only=False):
         if not (path or goto):  # special case for ThisPC
             return False
-        tests = self.view.settings().get('dired_hidden_files_patterns', ['.*'])
-        if isinstance(tests, str):
-            tests = [tests]
-        if any(fnmatch.fnmatch(filename, pattern) for pattern in tests):
-            return True
-        if sublime.platform() != 'windows':
-            return False
-        # check for attribute on windows:
-        try:
-            attrs = ctypes.windll.kernel32.GetFileAttributesW(join(path, goto, filename))
-            assert attrs != -1
-            result = bool(attrs & 2)
-        except (AttributeError, AssertionError):
-            result = False
-        return result
-
-    def try_listing_directory(self, path):
+        show_hidden = self.show_hidden
+        show_excluded = self.view.settings().get('dired_show_excluded_files', True)
+        is_hidden = False
+        filename = item['file']
+        fullpath = join(path, goto, filename)
+        is_dir = item['is_dir']
+        result = lambda: is_hidden if not get_is_dir else [is_hidden, is_dir]
+        if dirs_only and not is_dir:
+            return result()
+        if not is_hidden and not show_excluded:
+            if is_dir:
+                tests = self.view.settings().get('folder_exclude_patterns', [])
+                if any(fnmatch.fnmatch(filename, p) for p in tests):
+                    is_hidden = True
+            else:
+                tests = self.view.settings().get('file_exclude_patterns', [])
+                if any(fnmatch.fnmatch(filename, p) for p in tests):
+                    is_hidden = True
+        if not is_hidden and not show_hidden:
```

Comment:
> traverse_tree has neither checks, you meant try_listing_directory only, right?
> would be better to have the is_hidden/is_excluded check all at one place

`traverse_tree` [calls](https://github.com/aziz/SublimeFileBrowser/blob/master/dired.py?expand=1#L253) `try_listing_directory`, so `traverse_tree` do not need to check this itself.

The best is if hidden files are supposed to be _visible_ (due to `self.show_hidden`) then we do _not need_ to call `is_hidden` — it is important for performance, esp. on Windows, and in case of many files.

---

You did not change `prepare_filelist` method.

---

Line 434 seems weird: `bool` returns either True or False — why `if`? would be `is_hidden = bool(attrs & 2)`

---

You forgot to define which exception must be caught — `except (AttributeError, AssertionError):` (it is important in this case).

---

Cyclomatic complexity of `is_hidden` method is too high (18 if I count right), please try to keep it lower (11 is a desirable maximum).
If you don’t know: every `lambda`, `if`, `elif`, `try`, `and`, `or`, `assert` gives 1; you can use [radon](https://pypi.python.org/pypi/radon) to count automatically.
You can merge global (either folder or file) and dired patterns instead of check them separately.